projects/valtimo/zgw/src/lib/modules/zaakdetails/components/document-objecten-api-sync/document-objecten-api-sync.component.ts

```diff
@@ -0,0 +1,195 @@
+  SelectItem,
+  SelectModule,
+  SpinnerModule,
+  ValtimoCdsModalDirectiveModule,
+} from '@valtimo/components';
+import {TranslateModule} from '@ngx-translate/core';
+import {BehaviorSubject, map, Observable, Subscription, switchMap, tap} from 'rxjs';
+import {CommonModule} from '@angular/common';
+import {ButtonModule, CheckboxModule, ModalModule} from 'carbon-components-angular';
+import {DocumentObjectenApiSyncService} from '../../services';
+import {DocumentObjectenApiSync} from '../../models';
+import {DocumentDefinition} from '@valtimo/document';
+
+@Component({
+  selector: 'valtimo-document-objecten-api-sync',
+  templateUrl: './document-objecten-api-sync.component.html',
+  styleUrls: ['./document-objecten-api-sync.component.scss'],
+  standalone: true,
+  imports: [
+    CommonModule,
+    FormsModule,
+    TranslateModule,
+    SpinnerModule,
+    FormModule,
+    InputModule,
+    SelectModule,
+    ValtimoCdsModalDirectiveModule,
+    CheckboxModule,
+    ReactiveFormsModule,
+    ButtonModule,
+    ModalModule,
+  ],
+})
+export class DocumentObjectenApiSyncComponent implements OnInit, OnDestroy {
+  readonly loading$ = new BehaviorSubject<boolean>(true);
+  readonly documentDefinitionName$: Observable<string> = this.route.params.pipe(
+    map(params => params.name || '')
+  );
+  private readonly documentDefinition$: Observable<DocumentDefinition> =
+    this.documentDefinitionName$.pipe(
+      switchMap(documentDefinitionName =>
+        this.documentObjectenApiSyncService.getDocumentDefinition(documentDefinitionName)
+      )
+    );
+  readonly documentObjectenApiSync$ = new BehaviorSubject<DocumentObjectenApiSync>(null);
+  readonly objectManagementConfigurationItems$: Observable<Array<SelectItem>> =
+    this.documentObjectenApiSyncService.getObjectManagementConfigurations().pipe(
+      map(results =>
+        results?.map(configuration => ({
+          id: configuration.id,
+          text: configuration.title,
+        }))
+      )
+    );
+  readonly modalShowing$ = new BehaviorSubject<boolean>(false);
+  public readonly formGroup = new FormGroup({
+    objectManagementConfigurationId: new FormControl('', Validators.required),
+    enabled: new FormControl(true),
+  });
+
+  public get objectManagementConfigurationId(): AbstractControl<string> {
+    return this.formGroup.get('objectManagementConfigurationId');
+  }
+
+  public get enabled(): AbstractControl<boolean> {
+    return this.formGroup.get('enabled');
+  }
+
+  public readonly valid$ = new BehaviorSubject<boolean>(false);
+  private _subscriptions = new Subscription();
+
+  constructor(
+    private readonly route: ActivatedRoute,
+    private readonly documentObjectenApiSyncService: DocumentObjectenApiSyncService
+  ) {}
+
+  public ngOnInit(): void {
+    this.loadDocumentenObjectenApiSync();
+  }
+
+  public ngOnDestroy(): void {
+    this._subscriptions.unsubscribe();
+  }
+
+  public loadDocumentenObjectenApiSync(): void {
+    this._subscriptions.add(
```

Comment:
No need to add this to subscriptions. It's a HTTP call observable which already quits after emitting once